src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -257,24 +541,20 @@ ConsumerBufferPtr StorageRabbitMQ::popReadBuffer(std::chrono::milliseconds timeo
 
 ConsumerBufferPtr StorageRabbitMQ::createReadBuffer()
 {
-    if (update_channel_id)
-        next_channel_id += num_queues;
-    update_channel_id = true;
-
     ChannelPtr consumer_channel = std::make_shared<AMQP::TcpChannel>(connection.get());
 
     return std::make_shared<ReadBufferFromRabbitMQConsumer>(
-        consumer_channel, event_handler, exchange_name, routing_keys,
-        next_channel_id, log, row_delimiter, bind_by_id, num_queues,
-        exchange_type, local_exchange_name, stream_cancelled);
+        consumer_channel, setup_channel, event_handler, consumer_exchange, ++consumer_id,
+        unique_strbase, queue_base, log, row_delimiter, hash_exchange, num_queues,
+        deadletter_exchange, stream_cancelled);
 }
 
 
 ProducerBufferPtr StorageRabbitMQ::createWriteBuffer()
 {
     return std::make_shared<WriteBufferToRabbitMQProducer>(
-        parsed_address, global_context, login_password, routing_keys[0], local_exchange_name,
-        log, num_consumers * num_queues, bind_by_id, use_transactional_channel,
+        parsed_address, global_context, login_password, routing_keys, exchange_name, exchange_type,
+        producer_id.fetch_add(1), unique_strbase, persistent, wait_confirm, log,
```

Comment:
`producer_id` just a unique number?

Reply:
kind of, serial number to represent a channel id